src/Core/Domain/Manufacturer/VO/ManufacturerId.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Manufacturer\VO;
+
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\ManufacturerException;
+
+class ManufacturerId
+{
+
+    /**
+     * @var int
+     */
+    private $value;
+
+    /**
+     * ManufacturerId constructor.
+     *
+     * @param $manufacturerId
+     *
+     * @throws ManufacturerException
+     */
+    public function __construct($manufacturerId)
+    {
+        if (!is_numeric($manufacturerId) || $manufacturerId <= 0) {
+            throw new ManufacturerException(
+                sprintf('Invalid manufacturer id: %s', var_export($manufacturerId, true))
+            );
+        }
+
+        $this->value = (int) $manufacturerId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getValue()
```

Comment:
do we need a getValue()? I'd implement _toString() instead :)

Reply:
You can have both, the difference being that `getValue()` is **explicit** and `__toString()` is **magical**. Magic behavior is bad for maintenance.

Reply:
And:
> Warning You cannot throw an exception from within a __toString() method. Doing so will result in a fatal error.